### PR TITLE
DOC: Improve "random.generator.shuffle" docs page

### PR DESCRIPTION
**Pull request by bhavukkalra**
This PR is made in response to changes requested in issue #21034 . The requested changes seems very logical and Doable as well.
Currently Making this as a Draft as I am getting issues seeing changes reflect on the generated HTML pages after running 
`make html`.

### CODE REVIEW

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -4571,12 +4571,24 @@ cdef class Generator:
...
+        permuted
+        permutation
+
         Examples
         --------
         >>> rng = np.random.default_rng()
         >>> arr = np.arange(10)
         >>> rng.shuffle(arr)
         >>> arr
+        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```
Comment (rkern) [inline numpy/random/_generator.pyx:4590]: Did you mean to put this before the `rng.shuffle(arr)`?

Comment (rkern) [reply numpy/random/_generator.pyx:4590]: Be sure that the output matches. `arr` is not a list, so it won't look like this.

**Review by rkern — COMMENTED** [review]
A couple of minor fixes.

File: numpy/random/_generator.pyx (hunk 1 above)
Comment (bhavukkalra) [reply numpy/random/_generator.pyx:4590]: It is this way on the Official DOCS as well. [numpy.random.generator](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.shuffle.html).

But I guess this is what this PR is for(To fix it and Add new things)

**Comment by bhavukkalra** [comment]
Made the Suggested changes. (Below preview of the Page)

file_changes

**Review by rkern — APPROVED** [review]
Thanks! LGTM!

**Comment by rossbar** [comment]
Thanks @bhavukkalra - do you also want to add a ref to the handling-the-axis-param doc to the `permuted` docstring as well? This was suggested in the original issue and I think it's a good idea - might as well do it here so we can close the original issue as having been entirely addressed.

**Comment by bhavukkalra** [comment]
Made the suggested changes @rossbar

**Comment by charris** [comment]
Thanks @bhavukkalra .

**Pull request merged**